playwright/tests/parallel-3/liveblog-ad-limit.spec.ts

```diff
@@ -12,70 +12,89 @@ const desktopBreakpoint = breakpoints.filter(
 	({ breakpoint }) => breakpoint === 'desktop',
 )[0] as unknown as BreakpointSizes;
 
-/**
- * TODO e2e flakey test
- * - sometimes window.mockLiveUpdate is not available because the article does not switch to 'live'
- * - sometimes the maxAdSlots is 7 and not 8
- */
-
 const maxAdSlots = 8;
 
-const addNewBlocks = async (page: Page) => {
+const addAndAwaitNewBlocks = async (page: Page, blockContent: string) => {
 	// scroll to the top so we get a toast to click on
 	await page.evaluate(() => window.scrollTo(0, 0));
-	await page.evaluate(() => {
+	await page.evaluate((blockContent) => {
 		// @ts-expect-error -- browser land
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-call -- browser land
 		window.mockLiveUpdate({
 			numNewBlocks: 7,
 			html: `
-					<p style="height:1000px;" class="pending block">New block</p>
-					<p style="height:1000px;" class="pending block">New block</p>
-					<p style="height:1000px;" class="pending block">New block</p>
-					<p style="height:1000px;" class="pending block">New block</p>
-					<p style="height:1000px;" class="pending block">New block</p>
-					<p style="height:1000px;" class="pending block">New block</p>
-					<p style="height:1000px;" class="pending block">New block</p>
-				`,
+				<p style="height:1000px;" class="pending block">${blockContent}</p>
+				<p style="height:1000px;" class="pending block">New block</p>
+				<p style="height:1000px;" class="pending block">New block</p>
+				<p style="height:1000px;" class="pending block">New block</p>
+				<p style="height:1000px;" class="pending block">New block</p>
+				<p style="height:1000px;" class="pending block">New block</p>
+				<p style="height:1000px;" class="pending block">New block</p>
+			`,
 			mostRecentBlockId: 'abc',
 		});
-	});
+	}, blockContent);
 	// click on the toast to insert and scroll to the new blocks
 	await page.getByRole('button', { name: '7 new updates' }).click();
+
+	// Here, we force the test to wait until the blocks are visible on the page before counting the ad slots
+	const newBlock = page.getByText(blockContent).first();
+	await expect(newBlock).toBeVisible();
 };
 
 const countInlineSlots = (page: Page) =>
 	page.locator('#liveblog-body .ad-slot--liveblog-inline').count();
 
-test.describe('Ad slot limits', () => {
+test.describe.serial('Ad slot limits', () => {
```

Comment:
Nice detective work @emma-imber ! I'm surprised that we needed to add serial as each test should already be isolated ?: https://playwright.dev/docs/browser-contexts

Reply:
Yeah I thought the same! But when I remove it the tests don't pass. Not 100% sure what's going on but it seems we need it for the tests to work